src/v/cluster/controller_stm.cc

```diff
@@ -64,6 +67,10 @@ controller_stm::maybe_make_snapshot(ssx::semaphore_units apply_mtx_holder) {
         vlog(clusterlog.warn, "skipping snapshotting, feature not enabled");
         co_return std::nullopt;
     }
+    if (!_snapshot_enabled()) {
```

Comment:
Wondering if we need similar checks in apply_snapshot() code path.. for example: when snapshots are enabled for sometime and then disabled (may be due to a bug), leader installs the old snapshots during recovery but we may not want to apply them if the config is disabled and instead hydrate from the log?

Reply:
i think i agree, if the property is disabled we should ignore snapshot when bootstrapping